cpp/src/arrow/table.h

```diff
@@ -293,8 +291,12 @@ class ARROW_EXPORT TableBatchReader : public RecordBatchReader {
   void set_chunksize(int64_t chunksize);
 
  private:
-  class TableBatchReaderImpl;
-  std::unique_ptr<TableBatchReaderImpl> impl_;
```

Comment:
Why did you remove the pimpl? Is it just to remove one layer of indirection?

Reply:
Yes, it's no longer necessary